CMakeLists.txt

```diff
@@ -149,7 +149,7 @@ if (CMAKE_C_COMPILER_ID MATCHES "(Clang|Gnu|GNU)")
   # TODO: Set for MSVC and other compilers.
   # TODO: Set -Werror when the build is clean.
 
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -std=gnu99")
```

Comment:
We've actually been bitten by gnu extensions not working on some compilers/platforms in the past, so I'm kind of inclined to think we want c99, not gnu99. However, I'm not sure if that will break anything.

Tests suggest -std=c99 should be OK, but adding -pedantic spits out a bunch of warnings where we are not fully c99 compliant. We should probably fix that, but it gets a bit nasty because in some cases the way we are currently doing it will work for many Windows compilers that are not fully C99 compliant.

I think we probably want to make this -std=c99, and add -pedantic in another pull request that fixes all the non-C99 stuff, possibly adding conditional #define fixes for almost C99 compilers that we want to keep supporting.

Reply:
Sure, thanks for the review @dbaarda !

You're indeed right, I will update this PR with '-std=c99'

Reply:
Thanks for that, merging it now.